app/views/memberships/show.html.haml

```diff
@@ -8,3 +8,16 @@
   - if @membership.phone_number
     %p= @membership.phone_number
   %p= mail_to @membership.company_email, @membership.company_email
+  = form_for @membership, url: membership_path(@membership) do |f|
+    - if @membership.status.to_i == 2
+      - chosen = 2
+    - elsif @membership.status.to_i == 3
+      - chosen = 3
+    - else
+      - chosen = 1
```

Comment:
A bit too much logic in the view imo. I would suggest extracting this to a helper class.